test/box-luatest/gh_7846_crash_on_replica_id_alter_test.lua

```diff
@@ -0,0 +1,57 @@
+local server = require('test.luatest_helpers.server')
+local t = require('luatest')
+local urilib = require('uri')
+local fiber = require('fiber')
+
+local g = t.group('before_replace alter')
+
+g.test_replica_id_inc = function()
+    local server1 = server:new{
+        alias = 'server1',
+    }
+    server1:start()
+    server1:exec(function()
+        local trigger = function(_old, new)
+            return new:update({{'+', 1, 10}})
+        end
+        box.space._cluster:before_replace(trigger)
+    end)
+    local uri = urilib.parse(server1.net_box_uri)
+    local server2 = server:new{
+        alias = 'server2',
+        box_cfg = {replication = uri.unix}
+    }
+    server2:start()
+
+    local msg = server1:grep_log('Replica ID is changed by a trigger', 1024)
+    t.assert_not_equals(msg, nil)
+    server2:drop()
+    server1:drop()
+end
+
+g.test_replica_id_nullify = function()
+    local server1 = server:new{
+        alias = 'server1',
+    }
+    server1:start()
+    server1:exec(function()
+        local trigger = function(_old, _new)
+            return nil
+        end
+        box.space._cluster:before_replace(trigger)
+    end)
+    local uri = urilib.parse(server1.net_box_uri)
+    local server2 = server:new{
+        alias = 'server2',
+        box_cfg = {replication = uri.unix}
+    }
+    -- We cannot wait until ready because server2 will not be started at all,
+    -- so let's just sleep here.
+    server2:start({wait_for_readiness=false})
+    fiber.sleep(1)
+
+    local msg = server1:grep_log('Replica ID is changed by a trigger', 1024)
```

Comment:
Relying on 1 second being enough doesn't look good - 1) sometimes it won't be enough for sure, 2) most of the time it is too long. I would rather replace it with `t.helpers.retrying()` where you would wait on the log entry appearance. Also I am afraid that `server2:cleanup()` if done before the server is terminated might leave the server running. Wouldn't `:drop()` work?

Reply:
`drop()` does not work because `stop()` method, which is called before `cleanup()`, throws an error, because the server was not started. So `cleanup()` is the best option I've found.

Reply:
Replaced sleep with retry